Sources/MapboxDirections/RouteOptions.swift

```diff
@@ -203,7 +203,7 @@ open class RouteOptions: DirectionsOptions {
         }
         
         if waypoints.first(where: { $0.targetCoordinate != nil }) != nil {
-            let targetCoordinates = waypoints.map { $0.targetCoordinate?.requestDescription ?? "" }.joined(separator: ";")
+            let targetCoordinates = waypoints.filter { $0.separatesLegs == true }.map { $0.targetCoordinate?.requestDescription ?? "" }.joined(separator: ";")
```

Comment:
Nit: this property isn’t optional:

```suggestion
            let targetCoordinates = waypoints.filter { $0.separatesLegs }.map { $0.targetCoordinate?.requestDescription ?? "" }.joined(separator: ";")
```